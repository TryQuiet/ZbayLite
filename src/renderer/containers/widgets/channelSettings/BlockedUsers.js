import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import moderationActions from '../../../store/handlers/moderationActions'
import BlockedUsers from '../../../components/widgets/channelSettings/BlockedUsers'
import channelSelectors from '../../../store/selectors/channel'
import contactsSelectors from '../../../store/selectors/contacts'
// import messagesSelectors from '../../../store/selectors/messages'
import usersSelector from '../../../store/selectors/users'
import { moderationActionsType } from '../../../../shared/static'

export const mapStateToProps = state => {
  return {
    blockedUsers: contactsSelectors.directMessages(channelSelectors.id(state))(state).blockedUsers,
    users: usersSelector.users(state)
  }
}

export const mapDispatchToProps = (dispatch, props) =>
  bindActionCreators(
    {
      unblockUser: publicKey =>
        moderationActions.epics.handleModerationAction({
          moderationType: moderationActionsType.UNBLOCK_USER,
          moderationTarget: publicKey
        })
    },
    dispatch
  )

export default connect(mapStateToProps, mapDispatchToProps)(BlockedUsers)
