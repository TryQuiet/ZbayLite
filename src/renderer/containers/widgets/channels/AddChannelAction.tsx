import React from 'react'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import * as R from 'ramda'

import AddChannelAction from '../../../components/widgets/channels/AddChannelAction'
import { actionCreators } from '../../../store/handlers/modals'

export const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      openCreateModal: actionCreators.openModal('createChannel')
    },
    dispatch
  )

export default R.compose(
  React.memo,
  connect(
    null,
    mapDispatchToProps
  )
)(AddChannelAction)
