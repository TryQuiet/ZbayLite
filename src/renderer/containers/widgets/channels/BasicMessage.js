import React from 'react'
import { connect } from 'react-redux'

import BasicMessage from '../../../components/widgets/channels/BasicMessage'
import channelSelectors from '../../../store/selectors/channel'

export const mapStateToProps = state => {
  const isOwner = channelSelectors.isOwner(state)
  // TODO fix selector once moderation messages work
  const channelModerators = channelSelectors.channelModerators(state)
  return {
    allowModeration: isOwner || channelModerators
  }
}

export default connect(mapStateToProps)(React.memo(BasicMessage))
