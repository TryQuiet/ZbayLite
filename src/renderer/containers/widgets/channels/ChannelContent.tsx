import React from 'react'
import { connect } from 'react-redux'
import * as R from 'ramda'
import { bindActionCreators } from 'redux'

import ChannelContent from '../../../components/widgets/channels/ChannelContent'
import channelSelectors from '../../../store/selectors/channel'
// import identitySelectors from '../../../store/selectors/identity'
import mentionsSelectors from '../../../store/selectors/mentions'
import mentionsHandlers from '../../../store/handlers/mentions'

export const mapStateToProps = state => ({
  inputState: channelSelectors.inputLocked(state),
  // signerPubKey: identitySelectors.signerPubKey(state),
  mentions: mentionsSelectors.mentions(state),
  channelId: channelSelectors.channelId(state)
})

export const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      removeMention: mentionsHandlers.epics.removeMention
      // sendInvitation: mentionsHandlers.epics.sendInvitation
    },
    dispatch
  )

  

export default R.compose(
  React.memo,
  connect(mapStateToProps, mapDispatchToProps)
)(ChannelContent)
