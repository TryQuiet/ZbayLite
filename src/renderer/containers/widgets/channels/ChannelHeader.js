import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import Immutable from 'immutable'

import ChannelHeader from '../../../components/widgets/channels/ChannelHeader'
import channelsHandlers from '../../../store/handlers/channels'
import notificationCenterHandlers from '../../../store/handlers/notificationCenter'

import channelSelectors from '../../../store/selectors/channel'
import contactsSelectors from '../../../store/selectors/contacts'
import identitySelectors from '../../../store/selectors/identity'
import notificationCenter from '../../../store/selectors/notificationCenter'

import { messageType, notificationFilterType } from '../../../../shared/static'

export const mapStateToProps = (state, props) => {
  const contact = contactsSelectors.contact(props.contactId)(state)
  return {
    channel: Immutable.fromJS({
      name: props.contactId === 'general' ? 'zbay' : contact.get('username'),
      address: props.contactId
    }),
    name: contact.get('username'),
    userAddress: identitySelectors.address(state),
    members: channelSelectors.channelParticipiants(state),
    showAdSwitch: !!contactsSelectors
      .messages(props.contactId)(state)
      .find(msg => msg.type === messageType.AD),
    mutedFlag:
      notificationCenter.channelFilterById(
        channelSelectors.data(state) ? channelSelectors.data(state).key : 'none'
      )(state) === notificationFilterType.MUTE
  }
}
export const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      updateShowInfoMsg: channelsHandlers.epics.updateShowInfoMsg,
      unmute: () =>
        notificationCenterHandlers.epics.setChannelsNotification(
          notificationFilterType.ALL_MESSAGES
        )
    },
    dispatch
  )
export default connect(mapStateToProps, mapDispatchToProps)(ChannelHeader)
