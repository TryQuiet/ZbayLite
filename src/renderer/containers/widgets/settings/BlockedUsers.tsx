import React from 'react'
import { connect } from 'react-redux'
import * as R from 'ramda'
import { bindActionCreators } from 'redux'

import usersSelector from '../../../store/selectors/users'
import notificationCenterHandlers from '../../../store/handlers/notificationCenter'
import BlockedUsersComponent from '../../../components/widgets/settings/BlockedUsers'
import notificationCenterSelector from '../../../store/selectors/notificationCenter'

export const mapStateToProps = state => ({
  users: usersSelector.users(state),
  blockedUsers: notificationCenterSelector.blockedUsers(state)
})

export const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      unblock: notificationCenterHandlers.epics.unblockUserNotification
    },
    dispatch
  )

export const BlockedUsers = props => {
  return <BlockedUsersComponent {...props} />
}

export default R.compose(connect(mapStateToProps, mapDispatchToProps))(
  BlockedUsers
)
