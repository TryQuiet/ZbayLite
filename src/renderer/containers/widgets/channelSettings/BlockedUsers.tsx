
import React from 'react'

import BlockedUsers from '../../../components/widgets/channelSettings/BlockedUsers'

const BlockedUsersContainer = () => {
  return (
    <BlockedUsers
      blockedUsers={[]}
      unblockUser={() => {}}
      users={{}}
    />
  )
}

export default BlockedUsersContainer
