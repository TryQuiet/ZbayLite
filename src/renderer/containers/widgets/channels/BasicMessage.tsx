import React from 'react'

import BasicMessage from '../../../components/widgets/channels/BasicMessage'
import { IBasicMessageProps } from '../../../components/widgets/channels/BasicMessage.d'

export const useBasicMessageData = () => {

  // const isOwner = useSelector(publicChannels.selectors.isOwner)

  return {
    allowModeration: false
    // allowModeration: isOwner
  }
}

export const BasicMessageContainer: React.FC<IBasicMessageProps> = ({
  message,
  setActionsOpen,
  actionsOpen,
  allowModeration
}
) => {
  allowModeration = useBasicMessageData().allowModeration

  return (
    <BasicMessage
      message={message}
      setActionsOpen={setActionsOpen}
      actionsOpen={actionsOpen}
      allowModeration={allowModeration}
    />
  )
}

export default BasicMessageContainer
