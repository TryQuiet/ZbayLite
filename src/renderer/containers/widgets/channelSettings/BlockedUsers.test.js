/* eslint import/first: 0 */
import BigNumber from 'bignumber.js'

import { mapDispatchToProps, mapStateToProps } from './BlockedUsers'

import { ChannelState } from '../../../store/handlers/channel'
import { initialState } from '../../../store/handlers/channels'
import { createChannel } from '../../../testUtils'
import { ChannelMessages } from '../../../store/handlers/messages'

import create from '../../../store/create'

const messages = [
  {
    createdAt: 1567683687,
    id: 'test-1',
    type: 6,
    message: 'test',
    publicKey: '03638eb7aaae341acf66db8b79c9b31e3627715d8bf1795b87e817bda45cc80d',
    sender: {
      replyTo: '',
      username: 'test123'
    }
  },
  {
    createdAt: 1567683647,
    id: 'test-1',
    type: 7,
    message: 'test',
    publicKey: '03638eb7aaae341acf66db8b79c9b31e3627715d8bf1795b87e817bda45cc80d',
    sender: {
      replyTo: '',
      username: 'test123'
    }
  }
]
const channelId = 'randomid'
const baseStore = {
  channel: {
    ...ChannelState,
    spentFilterValue: 38,
    id: channelId,
    shareableUri: 'testuri',
    members: new BigNumber(0),
    message: 'Message written in the input'
  },
  channels: {
    ...initialState,
    data: [createChannel(channelId)]
  },
  certificates: {
    usersCertificates: ['MIIBtjCCAVwCBgF6Urb0VDAKBggqhkjOPQQDAjASMRAwDgYDVQQDEwdaYmF5IENBMB4XDTIxMDYyODEzMDIzOVoXDTMwMTIzMTIzMDAwMFowgZcxgZQwFAYKKwYBBAGDjBsCARMGZGFtaWFuMD8GA1UEAxM4YXp2eTcybmV4NXZzeWd2NW1ma3M0c2YyN2pycWRoZXpueDIzYm1xeHE3YzdtajNibW5pc3hhaWQwOwYJKwYBAgEPAwEBEy5RbVpWZnhrNWpQb2tiVmlqODNhWVV4ZXRaeWF5bnZCVlRCaGZQUlBSM3hndVZLMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAErEDIPiIQsZ020N34a+KtVLURm+68lyq0T7eklOlE6G3matt9vLKadwM4pNzJRhRigO1W0MNH3uOrwG1o2UOOX6MdMBswDAYDVR0TBAUwAwIBAzALBgNVHQ8EBAMCAAYwCgYIKoZIzj0EAwIDSAAwRQIgAocVe47k6ok0wA9vFaMRUxiD7syVPh+d3YIKLHanj5UCIQCNiXIh8DKsfkSICqzG3LxWl15W5Rq7wVij1fdM6AXo7Q=='],
    ownCertificate: {
      certificate: 'MIIBtjCCAVwCBgF6Urb0VDAKBggqhkjOPQQDAjASMRAwDgYDVQQDEwdaYmF5IENBMB4XDTIxMDYyODEzMDIzOVoXDTMwMTIzMTIzMDAwMFowgZcxgZQwFAYKKwYBBAGDjBsCARMGZGFtaWFuMD8GA1UEAxM4YXp2eTcybmV4NXZzeWd2NW1ma3M0c2YyN2pycWRoZXpueDIzYm1xeHE3YzdtajNibW5pc3hhaWQwOwYJKwYBAgEPAwEBEy5RbVpWZnhrNWpQb2tiVmlqODNhWVV4ZXRaeWF5bnZCVlRCaGZQUlBSM3hndVZLMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAErEDIPiIQsZ020N34a+KtVLURm+68lyq0T7eklOlE6G3matt9vLKadwM4pNzJRhRigO1W0MNH3uOrwG1o2UOOX6MdMBswDAYDVR0TBAUwAwIBAzALBgNVHQ8EBAMCAAYwCgYIKoZIzj0EAwIDSAAwRQIgAocVe47k6ok0wA9vFaMRUxiD7syVPh+d3YIKLHanj5UCIQCNiXIh8DKsfkSICqzG3LxWl15W5Rq7wVij1fdM6AXo7Q==',
      privateKey: 'MIGTAgEAMBMGByqGSM49AgEGCCqGSM49AwEHBHkwdwIBAQQgUMBHFeLut2Tyr5T5hkEtnbh4vx1XSEx2fBTwv29FytagCgYIKoZIzj0DAQehRANCAASsQMg+IhCxnTbQ3fhr4q1UtRGb7ryXKrRPt6SU6UTobeZq2328spp3Azik3MlGFGKA7VbQw0fe46vAbWjZQ45f'
    }
  }
}
describe('Send message popover', () => {
  let store = null
  beforeEach(() => {
    jest.clearAllMocks()
    store = create({
      ...baseStore,
      messages: {
        [channelId]: {
          ...ChannelMessages,
          messages: [...messages]
        }
      }
    })
  })

  it('will receive right props', () => {
    const state = mapStateToProps(store.getState())
    expect(state).toMatchSnapshot()
  })

  it('will receive right actions', () => {
    const actions = mapDispatchToProps(x => x)
    expect(actions).toMatchSnapshot()
  })
})
