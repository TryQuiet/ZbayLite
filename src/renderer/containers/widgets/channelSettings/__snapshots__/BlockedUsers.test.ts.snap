// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Send message popover will receive right actions 1`] = `Object {}`;

exports[`Send message popover will receive right props 1`] = `
Object {
  "blockedUsers": Array [],
  "users": Object {},
}
`;
